packages/flow/lib/shared.d.ts

```diff
@@ -2,6 +2,19 @@
 export function isIterable<T>(value: T): T extends Iterable<T> ? true : false
 
 /** Returns the given value as a normalized generator. */
-export function getNormalizedGenerator<T>(value: T): (
-	<T>(value: T) => AsyncGenerator<string, void, void>
+export function getNormalizedGenerator(value: unknown): (
+	(...args:Array<any>) => AsyncGenerator<string, void, void>
 )
+
+export type GetAppropriateFunctionBasedOnWhetherOrNotAGeneratorOfAnIterableWithTheForEachMethodIsPassed<T> = T extends HasForEachMethod
+	? Parameters<T["forEach"]>[0] : T extends Generator
```

Comment:
This type is the type that checks to see if an iterable has a for each method. If so the type of the first parameter of it's forEach method is the type. If it's a generator only the value is passed to the callback function for either iterate or `<For/>`